### DISCUSSION TITLE
How to contribute?

### DISCUSSION THREAD

**Discussion by tommcd** [pyenv/pyenv#2438] [Q&A] [answered]
Hi,

I'd like to submit a pull request, but having read [CONTRIBUTING.md](https://github.com/pyenv/pyenv/blob/master/CONTRIBUTING.md) I'm still unsure how to proceed as I think the information there is at a higher level than required by new contributors.

So I looked at GitHub Docs to understand how to contribute to a project and it suggested creating a pull request from a branch:
[creating-a-pull-request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request)

I have the following questions:

1. Since I don't have permission to create a new branch in the repository, I assume I need to fork the repository first?
2. Do I need to create a [New Issue](https://github.com/pyenv/pyenv/issues) to describe the (very small) change I want to propose or it is sufficient to describe it in the pull request?
3. I don't see any information in CONTRIBUTING.md on how to actually build and test the project, so I assume it is as simple as running make?
After installing "fish" all tests pass as shown below:

> make
PATH="./bats/bin:$PATH" test/run
 ✓ default version
 ...
  ✓ discovers version from pyenv-version-name
>
>178 tests, 0 failures
>
>cd plugins/python-build && /home/.../git/pyenv/pyenv/bats/bin/bats ${CI:+--tap} test
 ✓ not enough arguments for python-build
 ...
  ✓ git remote doesn't match
>
>126 tests, 0 failures
>

Sorry if the above is very basic, but I haven't contributed to a GitHub project before and I have searched [the discussions for 'contribute'](https://github.com/pyenv/pyenv/discussions?discussions_q=contribute) but haven't found any information.

Best regards,

Thomas

**Comment by native-api** [comment, marked as answer]
1. Fork the repository
2. Clone it locally
3. Create a new branch
4. Make the necessary changes
5. Push the changes to your fork as a branch
6. Create a pull request from the branch in your fork

This is standard Github M.O. so we didn't feel a need to duplicate these instructions and rather focused on the details specific for Pyenv.

---

There's no need to create an issue -- but it's advisable to do so if you're unsure if the problem you're having really is a problem with Pyenv.

---

For simple changes (like documentation wording), you can open the corresponding file at Github and press the "edit" button above its contents. Then there'll be an option to submit a PR with your changes.

> I don't see any information in CONTRIBUTING.md on how to actually build and test the project, so I assume it is as simple as running make?

Yes, `make test`.

Huh. I was sure this was covered by the documentation. Looks like it isn't! Worth adding.
